scylla-cdc-base/src/main/java/com/scylladb/cdc/model/master/Master.java

```diff
@@ -42,10 +46,52 @@ private GenerationId getGenerationId() throws InterruptedException, ExecutionExc
         }
     }
 
-    private boolean generationDone(GenerationMetadata generation, Set<TaskId> tasks) {
-        return generation.isClosed()
-                && !tasks.isEmpty()
```

Comment:
@haaawk Could not figure out why there was `&& !tasks.isEmpty()` - in this PR I removed it. Is it correct?

Reply:
I think it's not. For generation to be done it both has to be closed and all the tasks that read this generation have to finish reading relevant streams.

Reply:
I don't see it either. So it's probably fine to remove it.